llvm/lib/Target/AMDGPU/SIProgramInfo.cpp

```diff
@@ -199,3 +201,28 @@ const MCExpr *SIProgramInfo::getPGMRSrc2(CallingConv::ID CC,
 
   return MCConstantExpr::create(0, Ctx);
 }
+
+uint64_t SIProgramInfo::getFunctionCodeSize(const MachineFunction &MF) {
+  if (!CodeSizeInBytes.has_value()) {
+    const GCNSubtarget &STM = MF.getSubtarget<GCNSubtarget>();
+    const SIInstrInfo *TII = STM.getInstrInfo();
+
+    uint64_t CodeSize = 0;
+
+    for (const MachineBasicBlock &MBB : MF) {
+      for (const MachineInstr &MI : MBB) {
```

Comment:
Alignments not considered

Reply:
Elaborate?

Reply:
Just one of the ways this function is broken, it isn't accounting for the block alignments which will eventually be nop padded

Reply:
https://github.com/llvm/llvm-project/pull/127142